src/GitHub.VisualStudio/Menus/CopyLink.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.VisualStudio.Shell;
+using System;
+using System.ComponentModel.Composition;
+using System.Windows;
+using GitHub.Extensions;
+using GitHub.Services;
+
+namespace GitHub.VisualStudio.Menus
+{
+    [Export(typeof(IDynamicMenuHandler))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class CopyLink: LinkMenuBase, IDynamicMenuHandler
+    {
+        [ImportingConstructor]
+        public CopyLink([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+        }
+
+        public Guid Guid { get { return GuidList.guidContextMenuSet; } }
+        public int CmdId { get { return PkgCmdIDList.copyLinkCommand; } }
+
+        public void Activate()
+        {
+            if (!IsGitHubRepo())
+                return;
+
+            var link = GenerateLink();
+            if (link == null)
+                return;
+            Clipboard.SetText(link.AbsoluteUri);
```

Comment:
The documentation suggests `Clipboard.SetText` can only throw an `ArgumentNullException`. We've learned the hard way that's not true. For example, the [.NET source code](http://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/Clipboard.cs,139) shows it could throw a `SecurityException`.

Also, this interacts with COM and what happens in COM land, doesn't  stay in COM land but craps potential exceptions all over your best laid plans.

So we should probably wrap this in a `try/catch` so it doesn't crash the extension if adding to the Clipboard fails. In GitHub Desktop, we retry a couple times in case it's some weird intermittent failure. We don't have good data to know if it's possible that a retry would succeed though. So I'd just wrap it in a try/catch, write out a message, and the user can try again.